ce_comp.py

```diff
@@ -149,136 +169,162 @@ def main(args=None):
-            cprint(colored(point + " - " + str(thresholded_points_by_da[point])+" "), "cyan", attrs=['bold'])
+            #
+            error_count_tb = PrettyTable()
+            error_count_tb.field_names = ["Error", "Endpoints", "Avg Error"]
+            error_count_tb.add_row([error, count, error/count])
 
-        print("\nReliability Difference\n")
-        for point in sorted(thresholded_points_by_dr, key=thresholded_points_by_dr.get, reverse=True):
-            cprint(colored(point + " - " + str(thresholded_points_by_dr[point])+" "), "yellow", attrs=['bold'])
+            if output_format == "csv":
+                df.to_csv(save_path+tenant+"@"+date+"_report.csv", index=False, sep=",")
+            elif output_format == "html":
+                df.to_html(save_path+tenant+"@"+date+"_report.html", index=False)
+            else:
+                with open(save_path+tenant+"@"+date+"_report.json", "w") as fw:
+                    json.dump(endpoint_error_stats, fw)
+            log.info("Report saved successfully")
+
+            # table representation of the missing points
+            missing_points_tb = PrettyTable()
+            missing_points_tb.field_names = ["Name",
+                                             "Found in prod",
+                                             "Found in devel"]
+
+            for point in missing_points:
+                if point in prod_dict.keys():
+                    missing_points_tb.add_row([point, "Yes", "No"])
+                    log.info(key+" found in production but not in devel")
+                else:
+                    missing_points_tb.add_row([point, "No", "Yes"])
+                    log.info(key+" found in devel but not in production")
+
+            # terminal output for the calculated results
+            print("\n########################################")
+
+            print("\t\t\tDate: " + date+"\n")
+
+            print("Endpoints that were not found in both engines")
+
+            print(missing_points_tb)
+
+            print(error_count_tb)
+
+            print("\nEndpoints with a/r higher than the threshold"+"("+str(threshold)+").")
+            print("---------------------------------------------")
+
+            print("\nAvailability Difference\n")
+            for point in sorted(thresholded_points_by_da, key=thresholded_points_by_da.get, reverse=True):
+                cprint(colored(point + " - " + str(thresholded_points_by_da[point])+" "), "cyan", attrs=['bold'])
+
+            print("\nReliability Difference\n")
+            for point in sorted(thresholded_points_by_dr, key=thresholded_points_by_dr.get, reverse=True):
+                cprint(colored(point + " - " + str(thresholded_points_by_dr[point])+" "), "yellow", attrs=['bold'])
+
+        elif "results" in prod_data and "results" not in "devel_data":
+            log.critical("Comparison could not run for date: "+date+"\nDevel infrastructure didn't produce any results")
```

Comment:
Since you always log and print the same message in succession maybe use a method that does both? Or setup the logger to also output to the console